compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2655,6 +2655,10 @@ class Typer extends Namer
         tree.tpe <:< wildApprox(pt)
         readaptSimplified(Inliner.inlineCall(tree, pt))
       }
+      else if (tree.symbol.is(Macro, butNot = Inline)) {
```

Comment:
Why not check `Macro | Scala2x` ?

Reply:
The symbol does not have the `Scala2x` flag.

Reply:
Sounds like a bug ?

Reply:
Yes, I will investigate.

Reply:
The trick is that the `Scala2x` is on the owner of the macro. The flag `Scala2xMacro` was also wrong.